osu.Game/Screens/Select/BeatmapInfoWedge.cs

```diff
@@ -515,41 +542,58 @@ public InfoLabel(BeatmapStatistic statistic)
                 }
             }
 
-            private class DifficultyColourBar : Container
+            protected override void Dispose(bool isDisposing)
             {
-                private readonly StarDifficulty difficulty;
+                base.Dispose(isDisposing);
+                settingChangeTracker?.Dispose();
+            }
+        }
 
-                public DifficultyColourBar(StarDifficulty difficulty)
-                {
-                    this.difficulty = difficulty;
-                }
+        public class BufferedWedgeBackground : BufferedContainer
```

Comment:
So you would generalize it as a whole and not specific to the BeatmapInfo?

Reply:
Again not sure what you mean by that. In here I'm asking mostly for mechanical changes (split to own file, make `internal`, nest the `BufferedContainer` in a `CompositeDrawable` rather than using straight inheritance). No idea what "generalisation" would be in this context.

Reply:
To have a wedge background also for other components to use it but yeah we can let it tailored to the beatmap only